src/graph/node_iterator.c

```diff
@@ -42,7 +42,7 @@ Node *NodeIterator_Next(NodeIterator *iter) {
 
     // Advance to next block if current block consumed.
     if(iter->_block_pos >= NODEBLOCK_CAP) {
-        iter->_block_pos -= NODEBLOCK_CAP;
+        iter->_block_pos = 0;
```

Comment:
I think the original of this line was correct, as if `_step` is > 1 then we may not be interested in starting subsequent blocks on the 0th element.